Source/WebCore/css/StyleProperties.cpp

```diff
@@ -1621,6 +1621,19 @@ StringBuilder StyleProperties::asTextInternal() const
             shorthands.append(substitutionValue.shorthandPropertyId());
             value = substitutionValue.shorthandValue().cssText();
         } else {
+            // FIXME: could probably use matchingShorthandsForLonghand() instead of populating 'shorthands' manually.
+            switch (propertyID) {
+            case CSSPropertyCustom:
+            case CSSPropertyDirection:
+            case CSSPropertyUnicodeBidi:
+                // These are the only longhands not included in the 'all' shorthand.
+                break;
+            default:
+                ASSERT(propertyID >= firstCSSProperty);
+                ASSERT(propertyID < firstShorthandProperty);
```

Comment:
Looks like this assertion is failing in many tests; need to figure that out before landing this.

Reply:
Yeah, something like this:

```html
<!DOCTYPE html>
<textarea></textarea>
<script>
var input = document.querySelector('textarea');
input.focus();
input.selectionStart = 0;
document.execCommand('InsertHTML', false, '.');
</script>
```

somehow sets `animation: 0s ease 0s 1 normal none running none; background: rgba(0, 0, 0, 0) none repeat scroll 0% 0% / auto padding-box border-box; background-position: 0% 0%; border: 0px none rgb(0, 0, 0); border-block: 0px none rgb(0, 0, 0); border-block-color: rgb(0, 0, 0)�; border-block-end: 0px none rgb(0, 0, 0); border-block-start: 0px none rgb(0, 0, 0); border-block-style: none; border-block-width: 0px; border-bottom: 0px none rgb(0, 0, 0); border-color: rgb(0, 0, 0); border-image: none; border-inline: 0px none rgb(0, 0, 0)
                                                                                                                                                                                                                                                            �6; border-inline-color: rgb(0, 0, 0)�; border-inline-end: 0px none rgb(0, 0, 0); border-inline-start: 0px none rgb(0, 0, 0)
                                                                                                         �6; border-inline-style: noneal; border-inline-width: 0px; border-left: 0px none rgb(0, 0, 0)
                                                                                                                                                                                                      �6; border-radius: 0pxlace; border-right: 0px none rgb(0, 0, 0)
                                                                                                                                                                                                                                                                     �6; border-spacing: 0px 0px; border-style: none; border-top: 0px none rgb(0, 0, 0); border-width: 0px; column-rule: 0px none rgb(0, 0, 0); columns: auto auto; container: none; flex: 0 1 auto; flex-flow: row nowrap; font: normal normal normal normal 13px/normal "Noto Sans"; font-variant: normal; gap: normal normal; grid: none / none / none / row / auto / auto; grid-area: auto / auto / auto / auto���; grid-column: auto / auto; grid-row: auto / auto; grid-template: none / none / none; inset: auto; inset-block: auto; inset-inline: auto; list-style: outside none disc; margin: 0px; margin-block: 0px; margin-inline: 0px; mask: none; mask-position: 0% 0%; offset: auto none 0px auto 0deg / auto; outline: rgb(0, 0, 0) none 0px; overflow: visible; padding: 0px; padding-block: 0px; padding-inline: 0px; page-break-after: auto; page-break-before: auto; page-break-inside: auto; perspective-origin: 80.5px 9px; place-content: normal normal; place-items: normal normal; place-self: auto auto; scroll-margin: 0px; scroll-margin-block: 0pxo 0deg; scroll-margin-inline: 0px; scroll-padding: auto; scroll-padding-block: auto; scroll-padding-inline: auto; text-decoration: none; text-decoration-skip: auto; text-emphasis: none rgb(0, 0, 0); transform-origin: 80.5px 9px; transition: all 0s ease 0s; -webkit-background-size: auto; -webkit-border-image: none; -webkit-column-break-after: auto; -webkit-column-break-before: auto; -webkit-column-break-inside: auto; -webkit-mask-position: 0% 0%; -webkit-perspective: none; -webkit-text-decoration: none solid rgb(0, 0, 0)�; -webkit-text-orientation: mixed;` without expanding them into longhands!
This seems pretty broken but I guess it's unrelated. I guess I can just add a conditional with a FIXME.

Reply:
The reason for the assert failure was that `ComputedStyleExtractor::copyProperties()` copies both the longhands and the shorthands (without expanding them). This seems a regression from 7ee771049e3b77c5fa5f40ad9acc7703d6557c03, since previously in only copied computable properties, which are longhands.

Reply:
Good call; let’s fix that regression, too, maybe in a separate patch, and then we can add back the assertion. Attn: @dcrousso and @smfr.

Reply:
Created https://github.com/WebKit/WebKit/pull/5157